tests/src/Util/JSonLDTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * @copyright Copyright (C) 2020, Friendica
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Test\src\Util;
+
+use Friendica\Util\JsonLD;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * JsonLD utility test class
+ */
+class JsonLDTest extends TestCase
+{	
+	public function testFetchElementArrayNotFound()
+	{
+		$object = [];
+
+		$data = JsonLD::fetchElementArray($object, 'field');
+		$this->assertNull($data);
+	}
+
+	public function testFetchElementArrayFoundEmptyArray()
+	{
+		$object = ['field' => []];
+
+		$data = JsonLD::fetchElementArray($object, 'field');
+		$this->assertSame([], $data);
+	}
+
+	public function testFetchElementArrayFoundID()
+	{
+		$object = ['field' => ['value1', ['@id' => 'value2'], ['@id' => 'value3']]];
+
+		$data = JsonLD::fetchElementArray($object, 'field', '@id');
+		$this->assertSame(['value1', 'value2', 'value3'], $data);
+	}
+
+	public function testFetchElementArrayFoundArrays()
+	{
+		$object = ['field' => [['subfield11' => 'value11', 'subfield12' => 'value12'],
```

Comment:
What if one of these have an `@id`? The output become a array with a single element containing the value of the `@id` property. Is it intended? Why are the other elements discarded **by default**?

Reply:
This can't happen with the data, but I just changed the functionality a little bit.

Reply:
What data? The data you made up here? I'm talking about it because I encountered the case working with Mammoth messages. its was compacted alright and the result didn't have `@id` attributes on all the elements I was retrieving.

Reply:
Well, "Mammoth" seems to make stuff slightly different than other implementations. Hopefully with my small changes this works better.